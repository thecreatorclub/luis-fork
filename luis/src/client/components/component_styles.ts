import { initCss } from '@tomino/toolbelt';
import { ITheme } from '../config/themes';

export const { css } = initCss('luis-');

export const splitPane = (theme: ITheme) => css`
  /* name:split-container */
  background: ${theme.backgroundColor};
  color: ${theme.textColor}!important;

  .Resizer {
    background: #000;
    opacity: 0.2;
    z-index: 1;
    box-sizing: border-box;
    background-clip: padding-box;
  }
  .Resizer:hover {
    transition: all 2s easel;
  }
  .Resizer.horizontal {
    height: 11px;
    margin: -5px 0;
    border-top: 6px solid rgba(255, 255, 255, 0);
    border-bottom: 5px solid rgba(255, 255, 255, 0);
    cursor: row-resize;
    width: 100%;
  }
  .Resizer.horizontal:hover {
    border-top: 5px solid rgba(0, 0, 0, 0.5);
    border-bottom: 5px solid rgba(0, 0, 0, 0.5);
  }
  .Resizer.vertical {
    width: 11px;
    margin: 0 -5px;
    border-left: 5px solid rgba(255, 255, 255, 0);
    border-right: 5px solid rgba(255, 255, 255, 0);
    cursor: col-resize;
  }
  .Resizer.vertical:hover {
    border-left: 5px solid rgba(0, 0, 0, 0.5);
    border-right: 5px solid rgba(0, 0, 0, 0.5);
  }
  .SplitPane.horizontal {
    position: inherit!important as any;
  }
`;

export const content = css`
  /* name:content */
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  font-family: Lato;

  .m12 {
    margin: 12px !important;
  }
  .m6 {
    margin: 6px !important;
  }
`;

export const diff = css`
  max-height: 200px;
  overflow: auto;

  table {
    width: 100%;
    background: white;
    color: black;
  }

  .diff {
    border-collapse: collapse;
    white-space: pre-wrap;
  }

  .diff tbody {
    font-family: Courier, monospace;
    font-size: 11px;
  }
  .diff tbody th {
    background: #eed;
    font-size: 11px;
    font-weight: normal;
    border: 1px solid #bbc;
    color: #886;
    padding: 0.1em 0.4em 0.1em;
    text-align: right;
    vertical-align: top;
    width: 30px;
  }
  .diff thead {
    border-bottom: 1px solid #bbc;
    background: #efefef;
  }
  .diff thead th.texttitle {
    text-align: left;
    font-size: 11px;
  }
  .diff tbody td {
    padding-left: 0.4em;
    vertical-align: top;
  }
  .diff .empty {
    background-color: #ddd;
  }
  .diff .replace {
    background-color: #fd8;
  }
  .diff .delete {
    background-color: #e99;
  }
  .diff .skip {
    background-color: #efefef;
    border: 1px solid #aaa;
    border-right: 1px solid #bbc;
  }
  .diff .insert {
    background-color: #9e9;
  }
  .diff th.author {
    text-align: right;
    border-top: ˝1px solid #bbc;
    background: #efefef;
  }
`;
export const full = css`
  width: 100%;
  height: 100%;
`;

export const pane = (hideMenus: boolean) =>
  css`
    padding: 6px;
    position: absolute;
    overflow: auto;
    top: ${hideMenus ? '0px' : '42px'};
    bottom: 0;
    left: 0;
    right: 0;
  `;

export const timing = (color: string) =>
  css`
    /* name:timing */
    float: right;
    font-size: 10px;
    color: ${color};
    min-width: 40px;
    text-align: right;
  `;

export const floatShots = css`
  flex: 1 50px;
  min-width: 35px;
  text-align: left;
  margin-left: 5px;
`;
