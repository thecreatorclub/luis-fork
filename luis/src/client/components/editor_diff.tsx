import * as React from 'react';

function noop() {}

export interface Props {
  activeEditor?: string;
  className?: string;
  defaultValue?: string;
  editorDidMount?: (editor: any, monaco: any) => void;
  editorWillMount?: (monaco: any) => void;
  glyphMargin?: boolean;
  width?: string | number;
  height?: string | number;
  options?: any;
  requireConfig?: any;
  theme?: string;
  value: string;
  original: string;
  language: string;
  context?: any;
  onChange?: (val: string, e: any) => any;
}

export class MonacoDiffEditor extends React.Component<Props> {
  __current_value: string;
  __current_original: string;
  __prevent_trigger_change_event: boolean;
  containerElement: HTMLElement;
  editor: any;

  monitorResize: any;

  constructor(props: Props) {
    super(props);
    this.containerElement = undefined;
    this.__current_value = props.value;
    this.__current_original = props.original;
  }

  componentDidMount() {
    this.afterViewInit();
  }

  componentDidUpdate(prevProps: Props) {
    const context = this.props.context || window;
    if (
      this.props.value !== this.__current_value ||
      this.props.original !== this.__current_original
    ) {
      // Always refer to the latest value
      this.__current_value = this.props.value;
      this.__current_original = this.props.original;
      // Consider the situation of rendering 1+ times before the editor mounted
      if (this.editor) {
        this.__prevent_trigger_change_event = true;
        this.updateModel(this.__current_value, this.__current_original);
        this.__prevent_trigger_change_event = false;
      }
    }
    if (prevProps.language !== this.props.language) {
      context.monaco.editor.setModelLanguage(this.editor.getModel(), this.props.language);
    }
  }

  componentWillUnmount() {
    this.destroyMonaco();
  }

  editorWillMount(monaco: any) {
    const { editorWillMount } = this.props;
    editorWillMount(monaco);
  }

  editorDidMount(editor: any, monaco: any) {
    this.props.editorDidMount(editor, monaco);
    editor.onDidUpdateDiff((event: any) => {
      const value = editor.getValue();

      // Always refer to the latest value
      this.__current_value = value;

      // Only invoking when user input changed
      if (!this.__prevent_trigger_change_event) {
        this.props.onChange(value, event);
      }
    });
  }

  afterViewInit() {
    const context = this.props.context || window;
    if (context.monaco !== undefined) {
      this.initMonaco();
      return;
    }
    const { requireConfig } = this.props;
    const loaderUrl = requireConfig.url || 'vs/loader.js';
    const onGotAmdLoader = () => {
      if (context.__REACT_MONACO_EDITOR_LOADER_ISPENDING__) {
        // Do not use webpack
        if (requireConfig.paths && requireConfig.paths.vs) {
          context.require.config(requireConfig);
        }
      }

      // Load monaco
      context.require(['vs/editor/editor.main'], () => {
        this.initMonaco();
      });

      // Call the delayed callbacks when AMD loader has been loaded
      if (context.__REACT_MONACO_EDITOR_LOADER_ISPENDING__) {
        context.__REACT_MONACO_EDITOR_LOADER_ISPENDING__ = false;
        const loaderCallbacks = context.__REACT_MONACO_EDITOR_LOADER_CALLBACKS__;
        if (loaderCallbacks && loaderCallbacks.length) {
          let currentCallback = loaderCallbacks.shift();
          while (currentCallback) {
            currentCallback.fn.call(currentCallback.context);
            currentCallback = loaderCallbacks.shift();
          }
        }
      }
    };

    // Load AMD loader if necessary
    if (context.__REACT_MONACO_EDITOR_LOADER_ISPENDING__) {
      // We need to avoid loading multiple loader.js when there are multiple editors loading
      // concurrently, delay to call callbacks except the first one
      // eslint-disable-next-line max-len
      context.__REACT_MONACO_EDITOR_LOADER_CALLBACKS__ =
        context.__REACT_MONACO_EDITOR_LOADER_CALLBACKS__ || [];
      context.__REACT_MONACO_EDITOR_LOADER_CALLBACKS__.push({
        context: this,
        fn: onGotAmdLoader
      });
    } else if (typeof context.require === 'undefined') {
      const loaderScript = context.document.createElement('script');
      loaderScript.type = 'text/javascript';
      loaderScript.src = loaderUrl;
      loaderScript.addEventListener('load', onGotAmdLoader);
      context.document.body.appendChild(loaderScript);
      context.__REACT_MONACO_EDITOR_LOADER_ISPENDING__ = true;
    } else {
      onGotAmdLoader();
    }
  }

  updateModel(value: string, original: string) {
    const context = this.props.context || window;
    const { language } = this.props;
    const originalModel = context.monaco.editor.createModel(original, language);
    const modifiedModel = context.monaco.editor.createModel(value, language);
    this.editor.setModel({
      original: originalModel,
      modified: modifiedModel
    });
  }

  initMonaco() {
    const value = this.props.value !== null ? this.props.value : this.props.defaultValue;
    const { original, theme, options } = this.props;
    const context = this.props.context || window;
    if (this.containerElement && typeof context.monaco !== 'undefined') {
      // Before initializing monaco editor
      this.editorWillMount(context.monaco);
      this.editor = context.monaco.editor.createDiffEditor(this.containerElement, options);
      if (theme) {
        context.monaco.editor.setTheme(theme);
      }

      // monitor resize
      this.monitorResize = () => {
        this.editor.layout();
      };
      window.addEventListener('resize', this.monitorResize);

      // After initializing monaco editor
      this.updateModel(value, original);
      this.editorDidMount(this.editor, context.monaco);
    }
  }

  destroyMonaco() {
    if (typeof this.editor !== 'undefined') {
      this.editor.dispose();
    }

    window.removeEventListener('resize', this.monitorResize);
  }

  assignRef = (component: HTMLElement) => {
    this.containerElement = component;
  };

  render() {
    const { width, height } = this.props;
    const fixedWidth = width.toString().indexOf('%') !== -1 ? width : `${width}px`;
    const fixedHeight = height.toString().indexOf('%') !== -1 ? height : `${height}px`;
    const style = {
      width: fixedWidth,
      height: fixedHeight
    };

    return <div ref={this.assignRef} style={style} className="react-monaco-editor-container" />;
  }
}

MonacoDiffEditor.defaultProps = {
  width: '100%',
  height: '100%',
  original: null,
  value: null,
  defaultValue: '',
  language: 'javascript',
  theme: null,
  options: {},
  editorDidMount: noop,
  editorWillMount: noop,
  onChange: noop,
  requireConfig: {}
};

MonacoDiffEditor.displayName = 'MonacoDiffEditor';
