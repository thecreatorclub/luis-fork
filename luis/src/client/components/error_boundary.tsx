import * as React from 'react';
import { Segment } from 'semantic-ui-react';

export type State = { hasError: boolean; error?: Error; info?: React.ErrorInfo };

export class ErrorBoundary extends React.Component<any, State> {
  static displayName = 'ErrorBoundary';

  constructor(props: any) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error: Error, info: React.ErrorInfo) {
    // Display fallback UI
    this.setState({ hasError: true, error, info });
    // You can also log the error to an error reporting service
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return (
        <div>
          <Segment inverted attached="top" color="red">
            {this.state.error.message}
          </Segment>
          <Segment attached="bottom" color="red">
            <pre>{this.state.error.stack}</pre>

            <pre>{this.state.info.componentStack}</pre>
          </Segment>
        </div>
      );
    }
    return this.props.children;
  }
}
