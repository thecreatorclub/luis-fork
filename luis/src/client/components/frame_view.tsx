import * as React from 'react';
import { observer } from 'mobx-react';

import { css } from './component_styles';

type ComponentProps = {
  state?: Luis.State;
};

const frameStyle = css`
  border: 0px;
  width: 100%;
  height: 100%;
`;

const frameHolder = css`
  position: fixed;
  top: 140px;
  right: 0px;
  left: 400px;
  bottom: 0px;
  z-index: 100;
`;

export const FrameView = observer(({ state }: ComponentProps) => {
  return (
    <div
      className={frameHolder}
      style={{
        visibility: state.viewState.sView === 'react' ? 'visible' : 'hidden'
      }}
    >
      <iframe
        id="contentFrame"
        className={frameStyle}
        src={window.location.href.replace('?stories', '?story')}
      />
    </div>
  );
});

FrameView.displayName = 'StoryView';
