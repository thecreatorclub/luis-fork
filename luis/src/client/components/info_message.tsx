import * as React from 'react';
import { getState } from '../models/state_model';
import { ITheme } from '../config/themes';
import { css } from './component_styles';

const infoMessage = (theme: ITheme) => css`
  padding: 6px;
  color: ${theme.textColor};
`;

export const InfoMessage: React.FC = ({ children }) => {
  let state = getState();

  return <div className={infoMessage(state.theme)}>{children}</div>;
};
