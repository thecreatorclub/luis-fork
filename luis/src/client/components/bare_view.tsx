import * as React from 'react';

import { TopPanelSingle } from './top_panel_bare';

//@ts-ignore
import { StateModel } from '../models/state_model';
import { ErrorBoundary } from './error_boundary';
import { ThemedWrapper } from './themed_wrapper';
import { StoryComponent } from './story_component';

export type ComponentProps = {
  state?: Luis.State;
};

export const BareView = ({ state }: ComponentProps) => (
  <>
    {/* {!state.hideTestMenus */ false && <TopPanelSingle state={state} />}
    <ErrorBoundary>
      <ThemedWrapper state={state}>
        <StoryComponent state={state} />
      </ThemedWrapper>
    </ErrorBoundary>
  </>
);
