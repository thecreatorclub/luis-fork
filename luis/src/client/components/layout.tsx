import * as React from 'react';
import * as SplitPane from 'react-split-pane';

import { inject, observer } from 'mobx-react';

import { LeftPanel } from './left_panel';
import { RightPanel } from './right_panel';
import { splitPane, content } from './component_styles';

//@ts-ignore
import { StateModel } from '../config/state';
import { BareView } from './bare_view';

export type ComponentProps = {
  state?: Luis.State;
  localStorage: Storage;
};

export const Layout = inject('state')(
  observer(({ state, localStorage }: ComponentProps) => {
    // if (state.viewState.sView === 'live') {
    //   return <LiveSnapshots appState={state} />;
    // }
    if (state.viewState.bare) {
      return <BareView state={state} />;
    }
    return (
      <>
        {/* <FrameView state={state} /> */}
        <SplitPane
          className={splitPane(state.theme) + ' ' + content}
          split="vertical"
          minSize={100}
          defaultSize={parseInt(localStorage.getItem('luis-v-splitPos') || '280px', 10)}
          onDragStarted={() => (state.viewState.frame.style.pointerEvents = 'none')}
          onDragFinished={() => (state.viewState.frame.style.pointerEvents = 'auto')}
          onChange={(size: string) => localStorage.setItem('luis-v-splitPos', size)}
        >
          <LeftPanel state={state} />
          <RightPanel state={state} />
        </SplitPane>
        );
      </>
    );
  })
);
Layout.displayName = 'Layout';
