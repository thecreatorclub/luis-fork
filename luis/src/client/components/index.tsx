import * as ReactDOM from 'react-dom';
import * as React from 'react';
import { Provider } from 'mobx-react';

import { Layout } from './layout';
import { initState, RenderOptions } from '../models/state_model';
import { setupRouter } from '../config/router';
import { buildBridge } from '../../bridge';
import { ProxyStore } from '../models/proxy_store';
// import { setupHmr } from '../config/setup_hmr';

const state = initState();

// setupHmr();

export async function renderLuis(options: RenderOptions = {}) {
  let root = document.querySelector(options.root || '#react-root');
  if (!root) {
    root = document.createElement('div');
    root.id = options.root || 'react-root';
    document.body.appendChild(root);
  }

  // render application
  ReactDOM.render(<Luis options={options} />, root);
}

type Props = {
  options: RenderOptions;
};

export const Luis: React.FC<Props> = ({ options = {} }) => {
  state.renderOptions = options;

  buildBridge(options.report, options.snapshots);
  options.loadTests();

  // remember proxies
  if (options.proxies) {
    ProxyStore.init(options.proxies);
  }

  // create new router
  setupRouter(state);

  // make sure the left bar refreshes
  state.liveRoot.version++;

  return (
    <Provider state={state}>
      <Layout localStorage={localStorage} />
    </Provider>
  );
};
