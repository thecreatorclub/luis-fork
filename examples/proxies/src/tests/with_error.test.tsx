import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Foo } from '../components/foo';

describe('With Error', () => {
  function component() {
    return <Foo />;
  }

  it('renders correctly', () => {
    const wrapper = renderer.create(component());
    expect(wrapper).toMatchSnapshot('rendered');
  });

  it('shows difference', function() {
    expect('Actual Value').toEqual('Expected Value');
  });

  it('shows multiline difference', function() {
    expect({
      tomi: '1',
      bobo: '3',
      dodo: '5'
    }).toEqual({
      tomi: '1',
      bobo: '2',
      dodo: '5',
      lolo: 'rer'
    });
  });

  addStory({ component });
});
