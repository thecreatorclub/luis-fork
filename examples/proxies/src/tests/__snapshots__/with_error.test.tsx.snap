// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Error renders correctly: rendered 1`] = `
<div>
  <div
    className="ui red inverted top fixed menu"
  >
    <div
      className="item"
      onClick={[Function]}
    >
      Test
    </div>
    <a
      className="icon item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="plus icon"
        onClick={[Function]}
      />
      
    </a>
    <a
      className="icon item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="minus icon"
        onClick={[Function]}
      />
      
    </a>
  </div>
  <div
    style={
      Object {
        "paddingTop": "40px",
      }
    }
  >
    This is the component version ERROR: 
    0
  </div>
</div>
`;
